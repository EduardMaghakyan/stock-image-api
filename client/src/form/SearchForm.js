import React, { Component } from "react";

class SearchForm extends Component {
  state = {
    query: ""
  };
  handleSubmit = e => {
    e.preventDefault();
    this.props.handleSearch(e.target.query.value);
  };

  handleChange = e => {
    this.setState({
      query: e.target.value
    });
  };

  render() {
    const { query } = this.state;
    return (
      <div>
        <form
          onSubmit={this.handleSubmit}
          className="form-inline justify-content-center align-items-center"
        >
          <div className="form-group mx-sm-3 mb-2">
            <label htmlFor="query" className="sr-only">
              Keyword
            </label>
            <input
              type="text"
              id="query"
              name="query"
              value={query}
              onChange={this.handleChange}
              className="form-control"
              placeholder="Search"
            />
          </div>
          <button
            type="submit"
            className="btn btn-primary mb-2"
            id="search-submit"
          >
            Search
          </button>
        </form>
      </div>
    );
  }
}

export default SearchForm;
