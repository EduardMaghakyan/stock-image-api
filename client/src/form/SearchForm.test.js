import React from "react";
import ReactDOM from "react-dom";
import SearchForm from "./SearchForm";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });
describe("<SearchForm />", () => {
  // It should render.
  it("should render search form", () => {
    const wrapper = shallow(<SearchForm />);
    expect(wrapper.find("#query").props().value).toBe("");
    expect(wrapper.find("#search-submit").length).toBe(1);
  });
  // It allows to entery search query
  it("should render search form", () => {
    const wrapper = shallow(<SearchForm />);
    wrapper.find("#query").simulate("change", {
      target: { value: "My search query" }
    });
    expect(wrapper.find("#query").props().value).toBe("My search query");
  });

  // On click should call api with text entry
  it("should render search form", () => {
    const props = {
      handleSearch: jest.fn()
    };
    const wrapper = shallow(<SearchForm {...props} />);
    wrapper.find("form").simulate("submit", {
      preventDefault: () => {},
      target: { query: { value: "My search query" } }
    });
    expect(props.handleSearch).toHaveBeenCalledWith("My search query");
  });
  // Compare with sanpshot
  test("matches saved snapshot", () => {
    const wrapper = shallow(<SearchForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
