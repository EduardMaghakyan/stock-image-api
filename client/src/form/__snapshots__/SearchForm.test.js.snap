// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> matches saved snapshot 1`] = `
<div>
  <form
    className="form-inline justify-content-center align-items-center"
    onSubmit={[Function]}
  >
    <div
      className="form-group mx-sm-3 mb-2"
    >
      <label
        className="sr-only"
        htmlFor="query"
      >
        Keyword
      </label>
      <input
        className="form-control"
        id="query"
        name="query"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </div>
    <button
      className="btn btn-primary mb-2"
      id="search-submit"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
